nautobot/utilities/filters.py

```diff
@@ -427,11 +428,58 @@ class SearchFilter(MappedPredicatesFilterMixin, django_filters.CharFilter):
 #
 
 
-class BaseFilterSet(django_filters.FilterSet):
+class BaseFilterSetFilterForm(forms.Form):
+    """Base/default filter form class for BaseFilterSet."""
+
+    def is_valid(self):
+        """Extend forms.Form.is_valid() to treat any unknown form data entries as a potential validation error."""
+        result = super().is_valid()
+        for extra_key in set(self.data.keys()).difference(self.cleaned_data.keys()):
+            # If a given field was invalid, it will be omitted from cleaned_data; don't report extra errors
+            if extra_key not in self.errors:
+                if settings.STRICT_FILTERING:
+                    # Sure would be nice if we could instead do:
+                    # self.add_error(extra_key, "Unknown filter field") but extra_key isn't a field, so...
+                    self.add_error(None, f'Unknown filter field "{extra_key}"')
+                    result = False
+                else:
+                    logger.warning('%s: Unknown filter field "%s"', self.__class__.__name__, extra_key)
+
+        return result
+
+
+class BaseFilterSetOptions(FilterSetOptions):
     """
-    A base filterset which provides common functionality to all Nautobot filtersets.
+    Helper class for BaseFilterSetMetaclass.
 
-    Note that filtersets inheriting from this class MUST also inherit their Meta from BaseFilterSet.Meta or a subclass.
+    - Changes the default `_meta.form` to BaseFilterSetFilterForm.
+    - Ensures that `id` is always an included filter field
+    """
+
+    def __init__(self, options=None):
+        super().__init__(options)
+        if self.fields is not None and "id" not in self.fields:
+            self.fields = ("id", *self.fields)
+        self.form = getattr(options, "form", BaseFilterSetFilterForm)
```

Comment:
This is *chef's kiss*.

Reply:
Hope you think so about the reworked approach in the latest diffs too. :-)